libs/wingsdk/test/core/app.test.ts

```diff
@@ -0,0 +1,51 @@
+import { Construct } from "constructs";
+import { App } from "../../src/core/app";
+
+describe("new()", () => {
+  test("allows derived classes to override", () => {
```

Comment:
I spent a few minutes trying to think of edge cases but I couldn't come up with too much...

I suppose e.g. if I created an abstract resource that extends another abstract resource, it will also need a Wing plugin in order to reify it into a real class instance, but that probably makes sense.

Reply:
I am sure new edge cases will soon emerge